File: CS/MainForm.cs
using System;
using DevExpress.XtraSplashScreen;

namespace DXHtmlEditor {
    public partial class MainForm : DevExpress.XtraBars.Ribbon.RibbonForm {
        public MainForm() {
            InitializeComponent();
            dxHtmlEditorWebView.HtmlChanged += OnDxHtmlEditorWebViewHtmlChanged;
            dxHtmlEditorWebView.HtmlLoaded += OnDxHtmlEditorWebViewHtmlLoaded;
        }
        protected override void OnLoad(EventArgs e) {
            base.OnLoad(e);
            ShowOverlay();
            sourceMemoEdit.Text = "Hello DevExpress Html Editor!";
        }
        async void OnSourceMemoEditTextChanged(object sender, EventArgs e) {
            await dxHtmlEditorWebView.SetHtmlText(sourceMemoEdit.Text);
        }
        async void OnUndo(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            await dxHtmlEditorWebView.Undo();
        }
        async void OnRedo(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            await dxHtmlEditorWebView.Redo();
        }
        protected override void OnLookAndFeelChangedCore() {
            UpdateDxHtmlEditorWebViewTheme();
            base.OnLookAndFeelChangedCore();
        }
        bool UseDarkTheme {
            get { return DevExpress.Utils.Frames.FrameHelper.IsDarkSkin(GetActiveLookAndFeel()); }
        }
        async void UpdateDxHtmlEditorWebViewTheme() {
            string themeName = UseDarkTheme ? "dx.dark.css" : "dx.light.css";
            if(dxHtmlEditorWebView.CanSetTheme(themeName)) {
                ShowOverlay();
                await dxHtmlEditorWebView.SetTheme(themeName);
            }
        }
        async void OnDxHtmlEditorWebViewHtmlChanged(object sender, EventArgs e) {
            targetMemoEdit.Text = await dxHtmlEditorWebView.GetHtmlText();
        }
        void OnDxHtmlEditorWebViewHtmlLoaded(object sender, EventArgs e) {
            CloseOverlay();
        }
        IOverlaySplashScreenHandle overlayHandle;
        readonly OverlayWindowOptions options = new OverlayWindowOptions(opacity: 100d / 255);
        void ShowOverlay() {
            overlayHandle = overlayHandle ?? SplashScreenManager.ShowOverlayForm(layoutControl, options);
        }
        void CloseOverlay() {
            overlayHandle?.Close();
            overlayHandle = null;
        }
    }
}

File: VB/MainForm.vb
Imports System
Imports DevExpress.XtraSplashScreen

Namespace DXHtmlEditor

    Public Partial Class MainForm
        Inherits DevExpress.XtraBars.Ribbon.RibbonForm

        Public Sub New()
            InitializeComponent()
            AddHandler dxHtmlEditorWebView.HtmlChanged, AddressOf OnDxHtmlEditorWebViewHtmlChanged
            AddHandler dxHtmlEditorWebView.HtmlLoaded, AddressOf OnDxHtmlEditorWebViewHtmlLoaded
        End Sub

        Protected Overrides Sub OnLoad(ByVal e As EventArgs)
            MyBase.OnLoad(e)
            ShowOverlay()
            sourceMemoEdit.Text = "Hello DevExpress Html Editor!"
        End Sub

        Private Async Sub OnSourceMemoEditTextChanged(ByVal sender As Object, ByVal e As EventArgs)
            Await dxHtmlEditorWebView.SetHtmlText(sourceMemoEdit.Text)
        End Sub

        Private Async Sub OnUndo(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            Await dxHtmlEditorWebView.Undo()
        End Sub

        Private Async Sub OnRedo(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            Await dxHtmlEditorWebView.Redo()
        End Sub

        Protected Overrides Sub OnLookAndFeelChangedCore()
            UpdateDxHtmlEditorWebViewTheme()
            MyBase.OnLookAndFeelChangedCore()
        End Sub

        Private ReadOnly Property UseDarkTheme As Boolean
            Get
                Return DevExpress.Utils.Frames.FrameHelper.IsDarkSkin(GetActiveLookAndFeel())
            End Get
        End Property

        Private Async Sub UpdateDxHtmlEditorWebViewTheme()
            Dim themeName As String = If(UseDarkTheme, "dx.dark.css", "dx.light.css")
            If dxHtmlEditorWebView.CanSetTheme(themeName) Then
                ShowOverlay()
                Await dxHtmlEditorWebView.SetTheme(themeName)
            End If
        End Sub

        Private Async Sub OnDxHtmlEditorWebViewHtmlChanged(ByVal sender As Object, ByVal e As EventArgs)
            targetMemoEdit.Text = Await dxHtmlEditorWebView.GetHtmlText()
        End Sub

        Private Sub OnDxHtmlEditorWebViewHtmlLoaded(ByVal sender As Object, ByVal e As EventArgs)
            CloseOverlay()
        End Sub

        Private overlayHandle As IOverlaySplashScreenHandle

        Private ReadOnly options As OverlayWindowOptions = New OverlayWindowOptions(opacity:=100R / 255)

        Private Sub ShowOverlay()
            overlayHandle = If(overlayHandle, SplashScreenManager.ShowOverlayForm(layoutControl, options))
        End Sub

        Private Sub CloseOverlay()
            overlayHandle?.Close()
            overlayHandle = Nothing
        End Sub
    End Class
End Namespace
